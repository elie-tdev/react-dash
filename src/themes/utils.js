export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
