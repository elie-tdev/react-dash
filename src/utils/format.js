const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2,
});

const parseFloat = (str, val = 2) => {
  str = str && str.toString();
  str = str && str.indexOf('.') > -1 ? str.slice(0, str.indexOf('.') + val + 1) : str;
  return Number(str);
};

export const formatDollar = number => formatter.format(parseFloat(number, 2));
