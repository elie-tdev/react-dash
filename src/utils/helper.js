const maskAccountNumber = accNumber => {
  return accNumber && <>&bull;&bull;&bull;&bull; {accNumber.slice(accNumber.length - 4)}</>;
};

export default maskAccountNumber;
