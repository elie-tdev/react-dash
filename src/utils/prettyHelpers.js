export const prettyCurrency = (amount, currency = 'USD') => {
  if (amount == null) {
    return false;
  }
  if (amount < 0) {
    amount = 0;
  }
  return amount.toLocaleString('en', { style: 'currency', currency });
};

export const prettyDateTime = time => {
  if (time == null) {
    return false;
  }
  let d = new Date(time);
  return (
    d.toLocaleDateString('en', {
      month: '2-digit',
      day: '2-digit',
      year: '2-digit',
    }) +
    ' at ' +
    d.toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' }).replace(/^0(?:0:0?)?/, '')
  );
};

export const prettyDate = time => new Date(time).toLocaleDateString('en', {});

export const prettyTime = time =>
  new Date(time).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });

export const prettyHour = time =>
  new Date(time)
    .toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' })
    .replace(/^0(?:0:0?)?/, '');

export const prettyPhone = number =>
  number.substr(2).replace(/(\d{1,2})(\d{1})?(\d{1,3})?(\d{1,4})?/, function (_, p1, p2, p3, p4) {
    let output = '';
    if (p1) output = `${p1}`;
    if (p2) output += `${p2}-`;
    if (p3) output += `${p3}-`;
    if (p4) output += `${p4}`;
    return output;
  });

export const minimumDays = (date, numDays) => {
  const d = new Date(date);
  const totalNumDays = daysIntoYear(d) + numDays;
  return Boolean(daysIntoYear(d) >= totalNumDays || totalNumDays <= 365);
};

export const daysIntoYear = date => {
  return (
    (Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()) -
      Date.UTC(date.getFullYear(), 0, 0)) /
    24 /
    60 /
    60 /
    1000
  );
};

export const cleanAmountRegex = value => {
  return value.replace(/[^0-9]/g, '');
};

export const cleanAmount = value => {
  const pos = value.indexOf('.');
  let result;
  if (pos !== -1) {
    const part1 = value.substr(0, pos);
    const part2 = value.substr(pos + 1);
    result = cleanAmountRegex(part1) + '.' + cleanAmountRegex(part2);
  } else {
    result = cleanAmountRegex(value);
  }
  return result;
};
