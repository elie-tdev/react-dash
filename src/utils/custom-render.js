import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
