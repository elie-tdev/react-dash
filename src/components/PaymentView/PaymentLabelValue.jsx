import React from 'react';
import cn from 'classnames';
import { makeStyles, CardContent, Card, Box } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import { fade } from '@material-ui/core/styles/colorManipulator';

import useGlobalStyles from '@/themes/useGlobalStyles';
import { LabelValueColumn, PastDueToolTip } from '@/components';
import { formatDollar } from '@/utils/format';

const labelValueStyles = makeStyles(theme => ({
  containerStyle: {
    flex: 1,
    marginRight: '8px',
    '&:last-of-type': {
      marginRight: 0,
    },
    padding: '12px 12px 10px 16px',
    borderRadius: '8px',
    background: theme.palette.primary.main,
  },
  labelStyle: {
    marginBottom: '0',
    color: fade(theme.palette.primary.contrastText, 0.5),
  },
  valueStyle: {
    color: theme.palette.primary.contrastText,
  },
  grayContainerStyle: {
    background: 'rgba(245, 245, 247, 1)',
    height: '90px',
    flex: 1,
  },
  grayLabelStyle: {
    color: theme.palette.text.primary,
    opacity: 0.4,
  },
  grayValueStyle: {
    color: theme.palette.text.primary,
  },
  blueContainerStyle: {
    background: 'rgba(255, 255, 255, 0.1)',
    marginTop: '8px',
    width: '100%',
  },
  redContainerStyle: {
    background: theme.palette.error.lightBackground,
  },
  pastDueCard: {
    height: '90px',
    backgroundColor: theme.palette.error.lightBackground,
    borderRadius: '8px',
    flex: `0 0 calc(50% - 6px)`,
  },
  row: {
    width: '100%',
    justifyContent: 'space-between',
    flexWrap: 'wrap',
    '@media (max-width: 425px)': {
      display: 'block',
    },
  },
  grayBackground: {
    backgroundColor: theme.palette.common.grey[100],
  },
}));

const PaymentLabelValue = ({ paymentType, paymentValue = '500.00', paymentDate, dark }) => {
  const styles = labelValueStyles({ dark });
  const { t } = useTranslation();
  const globalStyles = useGlobalStyles();

  const blueLabelValue = {
    containerStyle: cn(styles.containerStyle, dark && styles.blueContainerStyle),
    labelStyle: cn(styles.labelStyle),
    valueStyle: cn(styles.valueStyle),
  };
  const grayLabelValue = {
    containerStyle: cn(styles.containerStyle, styles.grayContainerStyle),
    labelStyle: cn(styles.labelStyle, styles.grayLabelStyle),
    valueStyle: cn(styles.valueStyle, styles.grayValueStyle),
  };

  switch (paymentType) {
    case 'autoPay':
      return (
        <>
          <LabelValueColumn
            label={t('content.Payment Amount')}
            value={formatDollar(paymentValue)}
            classes={blueLabelValue}
          />
          <LabelValueColumn
            label={t('content.Payment Date')}
            value={paymentDate}
            classes={blueLabelValue}
          />
        </>
      );
    case 'autoPayEnrolled':
    case 'scheduledPayment':
    case 'pastDue':
      return (
        <Box className={cn(globalStyles.row, styles.row)}>
          <Card className={cn(globalStyles.card, styles.pastDueCard)} elevation={0}>
            <CardContent className={cn(globalStyles.row)}>
              <LabelValueColumn
                label={t('content.Past Due')}
                value={formatDollar(paymentValue)}
                variant="error"
                size="lg"
              />
              <PastDueToolTip />
            </CardContent>
          </Card>
          <Card
            className={cn(globalStyles.card, styles.pastDueCard, styles.grayBackground)}
            elevation={0}
          >
            <CardContent className={cn(globalStyles.row)}>
              <LabelValueColumn label={t('content.Amount Due')} value={paymentDate} size="lg" />
            </CardContent>
          </Card>
        </Box>
      );
    case 'paymentDue':
      return (
        <>
          <LabelValueColumn
            label={t('content.Payment Amount')}
            value={formatDollar(paymentValue)}
            classes={grayLabelValue}
          />
          <LabelValueColumn
            label={t('content.Payment Date')}
            value={paymentDate}
            classes={grayLabelValue}
          />
        </>
      );
    default:
      return null;
  }
};

export default PaymentLabelValue;
