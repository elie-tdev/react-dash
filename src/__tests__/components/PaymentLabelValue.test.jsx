import React from 'react';

import { render } from '@/utils/custom-render';
import PaymentLabelValue from '@/components/PaymentView/PaymentLabelValue';

describe('PaymentLabelValue Component', () => {
  test('Renders PaymentLabelValue', () => {
    const { container } = render(
      <PaymentLabelValue paymentType="paymentDue" paymentAmount="500" paymentDate="Jan 12, 2021" />,
    );
    expect(container).toMatchSnapshot();
  });

  test('Renders PaymentLabelValue if paymentType is autoPayEnrolled', () => {
    const { container } = render(
      <PaymentLabelValue
        paymentType="autoPayEnrolled"
        paymentAmount="500"
        paymentDate="Jan 12, 2021"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('Renders PaymentLabelValue if paymentType is not passed', () => {
    const { container } = render(
      <PaymentLabelValue paymentAmount="500" paymentDate="Jan 12, 2021" />,
    );
    expect(container).toMatchSnapshot();
  });
});
