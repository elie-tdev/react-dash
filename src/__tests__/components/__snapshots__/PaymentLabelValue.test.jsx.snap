// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentLabelValue Component Renders PaymentLabelValue 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-46 makeStyles-container-44 makeStyles-containerStyle-1 makeStyles-grayContainerStyle-4"
    data-testid="LabelValueColumn"
  >
    <h6
      class="MuiTypography-root makeStyles-labelStyle-2 makeStyles-grayLabelStyle-5 MuiTypography-h6"
    >
      Payment Amount
    </h6>
    <p
      class="MuiTypography-root makeStyles-value-41 makeStyles-value-45 makeStyles-valueSmall-43 makeStyles-valueStyle-3 makeStyles-grayValueStyle-6 MuiTypography-body1"
    >
      $500.00
    </p>
  </div>
  <div
    class="MuiBox-root MuiBox-root-48 makeStyles-container-44 makeStyles-containerStyle-1 makeStyles-grayContainerStyle-4"
    data-testid="LabelValueColumn"
  >
    <h6
      class="MuiTypography-root makeStyles-labelStyle-2 makeStyles-grayLabelStyle-5 MuiTypography-h6"
    >
      Payment Date
    </h6>
    <p
      class="MuiTypography-root makeStyles-value-41 makeStyles-value-47 makeStyles-valueSmall-43 makeStyles-valueStyle-3 makeStyles-grayValueStyle-6 MuiTypography-body1"
    >
      Jan 12, 2021
    </p>
  </div>
</div>
`;

exports[`PaymentLabelValue Component Renders PaymentLabelValue if paymentType is autoPayEnrolled 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-89 makeStyles-row-67 makeStyles-row-58"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-63 makeStyles-pastDueCard-57 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root makeStyles-row-67"
      >
        <div
          class="MuiBox-root MuiBox-root-95 makeStyles-container-93"
          data-testid="LabelValueColumn"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            Past Due
          </h6>
          <p
            class="MuiTypography-root makeStyles-value-90 makeStyles-value-94 makeStyles-valueLarge-91 MuiTypography-body1"
          >
            $500.00
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-63 makeStyles-pastDueCard-57 makeStyles-grayBackground-59 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root makeStyles-row-67"
      >
        <div
          class="MuiBox-root MuiBox-root-108 makeStyles-container-93"
          data-testid="LabelValueColumn"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            Amount Due
          </h6>
          <p
            class="MuiTypography-root makeStyles-value-90 makeStyles-value-107 makeStyles-valueLarge-91 MuiTypography-body1"
          >
            Jan 12, 2021
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PaymentLabelValue Component Renders PaymentLabelValue if paymentType is not passed 1`] = `<div />`;
